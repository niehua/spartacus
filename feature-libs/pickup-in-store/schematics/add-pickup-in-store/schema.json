{
  "$schema": "http://json-schema.org/schema",
  "$id": "PickupInStoreSchematics",
  "title": "Pickup in Store Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Pickup in Store feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Pickup in Store"]
    }
  },
  "required": []
}
