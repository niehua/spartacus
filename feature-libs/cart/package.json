{
  "name": "@spartacus/cart",
  "version": "4.1.0-next.0",
  "description": "",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "cart",
    "shopping",
    "shop",
    "saved cart",
    "quick order"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/feature-libs/cart",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "yarn clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "yarn --cwd ../../projects/schematics/ run clean && yarn clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^13.3.0",
    "@angular/common": "^13.3.0",
    "@angular/core": "^13.3.0",
    "@angular/forms": "^13.3.0",
    "@angular/router": "^13.3.0",
    "@ng-bootstrap/ng-bootstrap": "^11.0.0",
    "@ng-select/ng-select": "^8.0.0",
    "@ngrx/effects": "^13.0.2",
    "@ngrx/store": "^13.0.2",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/schematics": "4.1.0-next.0",
    "@spartacus/storefront": "4.1.0-next.0",
    "@spartacus/styles": "4.1.0-next.0",
    "@spartacus/user": "4.1.0-next.0",
    "bootstrap": "^4.6.1",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
