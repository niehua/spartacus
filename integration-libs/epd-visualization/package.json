{
  "name": "@spartacus/epd-visualization",
  "version": "4.1.0-next.0",
  "description": "SAP Enterprise Product Development Visualization integration library for Spartacus",
  "keywords": [
    "spartacus",
    "framework",
    "storefront",
    "epd-visualization",
    "2d",
    "3d",
    "visual",
    "spare",
    "part",
    "picking",
    "epd"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/integration-libs/epd-visualization",
  "license": "Apache-2.0",
  "exports": {
    ".": {
      "sass": "./_index.scss"
    }
  },
  "scripts": {
    "build:schematics": "yarn clean:schematics && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean:schematics": "../../node_modules/.bin/rimraf \"schematics/**/*.js\" \"schematics/**/*.js.map\" \"schematics/**/*.d.ts\"",
    "test:schematics": "yarn --cwd ../../projects/schematics/ run clean && yarn clean:schematics && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/schematics": "^13.3.0",
    "@angular/common": "^13.3.0",
    "@angular/core": "^13.3.0",
    "@angular/forms": "^13.3.0",
    "@angular/router": "^13.3.0",
    "@sapui5/ts-types-esm": "1.98.0",
    "@spartacus/cart": "4.1.0-next.0",
    "@spartacus/core": "4.1.0-next.0",
    "@spartacus/schematics": "4.1.0-next.0",
    "@spartacus/storefront": "4.1.0-next.0",
    "@spartacus/styles": "4.1.0-next.0",
    "bootstrap": "^4.6.1",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "schematics": "./schematics/collection.json"
}
