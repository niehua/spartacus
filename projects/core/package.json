{
  "name": "@spartacus/core",
  "version": "4.1.0-next.0",
  "description": "Spartacus - the core framework",
  "keywords": [
    "spartacus",
    "storefront",
    "angular",
    "framework"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/projects/core",
  "license": "Apache-2.0",
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular/common": "^13.3.0",
    "@angular/core": "^13.3.0",
    "@angular/platform-browser": "^13.3.0",
    "@angular/router": "^13.3.0",
    "@ngrx/effects": "^13.0.2",
    "@ngrx/router-store": "^13.0.2",
    "@ngrx/store": "^13.0.2",
    "angular-oauth2-oidc": "^13.0.1",
    "i18next": "^21.6.10",
    "i18next-http-backend": "^1.3.2",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
