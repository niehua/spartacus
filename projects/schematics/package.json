{
  "name": "@spartacus/schematics",
  "version": "4.1.0-next.0",
  "description": "Spartacus schematics",
  "keywords": [
    "spartacus",
    "schematics"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/projects/schematics",
  "license": "MIT",
  "main": "index.js",
  "scripts": {
    "build": "yarn clean && yarn --cwd ../../ generate:deps && ../../node_modules/.bin/tsc -p ./tsconfig.schematics.json",
    "clean": "../../node_modules/.bin/rimraf \"index.js*\" \"index.d.ts\" \"src/**/*.js\" \"src/**/*.js.map\" \"src/**/*.d.ts\"",
    "test": "yarn clean && ../../node_modules/.bin/jest --config ./jest.schematics.config.js"
  },
  "dependencies": {
    "@angular/pwa": "^13.3.0",
    "@nguniversal/express-engine": "^13.1.1",
    "semver": "^7.3.5",
    "ts-morph": "^9.1.0",
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular-devkit/core": "^13.3.0",
    "@angular-devkit/schematics": "^13.3.0",
    "@angular/compiler": "^13.3.0",
    "@schematics/angular": "^13.3.0",
    "jsonc-parser": "^3.0.0",
    "parse5": "^6.0.1",
    "typescript": "^4.6.4"
  },
  "ng-add": {
    "save": "devDependencies"
  },
  "ng-update": {
    "migrations": "./src/migrations/migrations.json",
    "packageGroup": [
      "@spartacus/core",
      "@spartacus/storefront",
      "@spartacus/styles",
      "@spartacus/user",
      "@spartacus/cds",
      "@spartacus/assets",
      "@spartacus/organization",
      "@spartacus/product",
      "@spartacus/storefinder",
      "@spartacus/checkout",
      "@spartacus/asm",
      "@spartacus/smartedit",
      "@spartacus/qualtrics",
      "@spartacus/cdc",
      "@spartacus/tracking",
      "@spartacus/cart",
      "@spartacus/order",
      "@spartacus/digital-payments",
      "@spartacus/epd-visualization",
      "@spartacus/setup"
    ]
  },
  "schematics": "./src/collection.json"
}
