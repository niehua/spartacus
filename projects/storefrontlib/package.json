{
  "name": "@spartacus/storefront",
  "version": "4.1.0-next.0",
  "keywords": [
    "spartacus",
    "storefront",
    "angular"
  ],
  "homepage": "https://github.com/SAP/spartacus",
  "repository": "https://github.com/SAP/spartacus/tree/develop/projects/storefrontlib",
  "license": "Apache-2.0",
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "peerDependencies": {
    "@angular/common": "^13.3.0",
    "@angular/core": "^13.3.0",
    "@angular/forms": "^13.3.0",
    "@angular/platform-browser": "^13.3.0",
    "@angular/router": "^13.3.0",
    "@angular/service-worker": "^13.3.0",
    "@ng-bootstrap/ng-bootstrap": "^11.0.0",
    "@ng-select/ng-select": "^8.0.0",
    "@ngrx/effects": "^13.0.2",
    "@ngrx/router-store": "^13.0.2",
    "@ngrx/store": "^13.0.2",
    "@spartacus/core": "4.1.0-next.0",
    "ngx-infinite-scroll": "^8.0.0",
    "rxjs": "^6.6.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
