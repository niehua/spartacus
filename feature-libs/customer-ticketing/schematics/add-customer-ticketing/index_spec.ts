/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  customerTicketingFeatureModulePath,
  CUSTOMER_TICKETING_FEATURE_NAME,
  LibraryOptions as SpartacusCustomerTicketingOptions,
  SpartacusOptions,
  SPARTACUS_CUSTOMER_TICKETING,
  SPARTACUS_SCHEMATICS,
} from '@spartacus/schematics';
import * as path from 'path';
import { peerDependencies } from '../../package.json';

const collectionPath = path.join(__dirname, '../collection.json');
const scssFilePath = 'src/styles/spartacus/customer-ticketing.scss';

describe('Spartacus Customer Ticketing schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_CUSTOMER_TICKETING,
    collectionPath
  );

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: SpartacusCustomerTicketingOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const customerTicketingFeatureOptions: SpartacusCustomerTicketingOptions = {
    ...libraryNoFeaturesOptions,
    features: [CUSTOMER_TICKETING_FEATURE_NAME],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      '../../projects/schematics/src/collection.json'
    );

    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'workspace',
        workspaceOptions
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        '@schematics/angular',
        'application',
        appOptions,
        appTree
      )
      .toPromise();
    appTree = await schematicRunner
      .runExternalSchematicAsync(
        SPARTACUS_SCHEMATICS,
        'ng-add',
        { ...spartacusDefaultOptions, name: 'schematics-test' },
        appTree
      )
      .toPromise();
  });

  describe('Without features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner
        .runSchematicAsync('ng-add', libraryNoFeaturesOptions, appTree)
        .toPromise();
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(customerTicketingFeatureModulePath)).toBeFalsy();
    });

    it('should install necessary Spartacus libraries', () => {
      const packageJson = JSON.parse(appTree.readContent('package.json'));
      let dependencies: Record<string, string> = {};
      dependencies = { ...packageJson.dependencies };
      dependencies = { ...dependencies, ...packageJson.devDependencies };

      for (const toAdd in peerDependencies) {
        // skip the SPARTACUS_SCHEMATICS, as those are added only when running by the Angular CLI, and not in the testing environment
        if (
          !peerDependencies.hasOwnProperty(toAdd) ||
          toAdd === SPARTACUS_SCHEMATICS
        ) {
          continue;
        }
        // TODO: after 4.0: use this test, as we'll have synced versions between lib's and root package.json
        // const expectedVersion = (peerDependencies as Record<
        //   string,
        //   string
        // >)[toAdd];
        const expectedDependency = dependencies[toAdd];
        expect(expectedDependency).toBeTruthy();
        // expect(expectedDependency).toEqual(expectedVersion);
      }
    });
  });

  describe('Customer Ticketing feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync('ng-add', customerTicketingFeatureOptions, appTree)
          .toPromise();
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(customerTicketingFeatureModulePath);
        expect(module).toMatchSnapshot();
      });

      describe('styling', () => {
        it('should create a proper scss file', () => {
          const scssContent = appTree.readContent(scssFilePath);
          expect(scssContent).toMatchSnapshot();
        });

        it('should update angular.json', async () => {
          const content = appTree.readContent('/angular.json');
          expect(content).toMatchSnapshot();
        });
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner
          .runSchematicAsync(
            'ng-add',
            { ...customerTicketingFeatureOptions, lazy: false },
            appTree
          )
          .toPromise();
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(customerTicketingFeatureModulePath);
        expect(module).toMatchSnapshot();
      });
    });
  });
});
