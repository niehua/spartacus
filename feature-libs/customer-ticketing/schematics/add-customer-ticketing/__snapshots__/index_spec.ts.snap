// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus Customer Ticketing schematics: ng-add Customer Ticketing feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { I18nConfig, provideConfig } from "@spartacus/core";
import { CustomerTicketingModule } from "@spartacus/customer-ticketing";
import { customerTicketingTranslationChunksConfig, customerTicketingTranslations } from "@spartacus/customer-ticketing/assets";
import { CustomerTicketingRootModule } from "@spartacus/customer-ticketing/root";

@NgModule({
  declarations: [],
  imports: [
    CustomerTicketingRootModule,
    CustomerTicketingModule
  ],
  providers: [provideConfig(<I18nConfig>{
    i18n: {
      resources: customerTicketingTranslations,
      chunks: customerTicketingTranslationChunksConfig,
    },
  })]
})
export class CustomerTicketingFeatureModule { }
"
`;

exports[`Spartacus Customer Ticketing schematics: ng-add Customer Ticketing feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CmsConfig, I18nConfig, provideConfig } from "@spartacus/core";
import { customerTicketingTranslationChunksConfig, customerTicketingTranslations } from "@spartacus/customer-ticketing/assets";
import { CustomerTicketingRootModule, CUSTOMER_TICKETING_FEATURE } from "@spartacus/customer-ticketing/root";

@NgModule({
  declarations: [],
  imports: [
    CustomerTicketingRootModule
  ],
  providers: [provideConfig(<CmsConfig>{
    featureModules: {
      [CUSTOMER_TICKETING_FEATURE]: {
        module: () =>
          import('@spartacus/customer-ticketing').then((m) => m.CustomerTicketingModule),
      },
    }
  }),
  provideConfig(<I18nConfig>{
    i18n: {
      resources: customerTicketingTranslations,
      chunks: customerTicketingTranslationChunksConfig,
    },
  })
  ]
})
export class CustomerTicketingFeatureModule { }
"
`;

exports[`Spartacus Customer Ticketing schematics: ng-add Customer Ticketing feature general setup styling should create a proper scss file 1`] = `
"@import "../../styles-config";
@import "@spartacus/customer-ticketing";
"
`;

exports[`Spartacus Customer Ticketing schematics: ng-add Customer Ticketing feature general setup styling should update angular.json 1`] = `
"{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "",
  "projects": {
    "schematics-test": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/schematics-test",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.scss",
              "src/styles/spartacus/customer-ticketing.scss"
            ],
            "scripts": [],
            "stylePreprocessorOptions": {
              "includePaths": [
                "node_modules/"
              ]
            }
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "2.5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "fileReplacements": [
                {
                  "replace": "src/environments/environment.ts",
                  "with": "src/environments/environment.prod.ts"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "browserTarget": "schematics-test:build:production"
            },
            "development": {
              "browserTarget": "schematics-test:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "schematics-test:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "main": "src/test.ts",
            "polyfills": "src/polyfills.ts",
            "tsConfig": "tsconfig.spec.json",
            "karmaConfig": "karma.conf.js",
            "inlineStyleLanguage": "scss",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "src/styles.scss",
              "src/styles/spartacus/customer-ticketing.scss"
            ],
            "scripts": [],
            "stylePreprocessorOptions": {
              "includePaths": [
                "node_modules/"
              ]
            }
          }
        }
      }
    }
  }
}"
`;
